vectordb_orm/backends/pinecone/pinecone.py

```diff
@@ -0,0 +1,256 @@
+from logging import info
+from re import match as re_match
+from typing import Type
+from uuid import uuid4
+
+import numpy as np
+import pinecone
+
+from vectordb_orm.attributes import AttributeCompare
+from vectordb_orm.backends.base import BackendBase
+from vectordb_orm.backends.pinecone.indexes import PineconeIndex
+from vectordb_orm.base import VectorSchemaBase
+from vectordb_orm.fields import EmbeddingField
+from vectordb_orm.results import QueryResult
+from vectordb_orm.attributes import OperationType
+
+
+class PineconeBackend(BackendBase):
+    max_fetch_size = 1000
+
+    def __init__(
+        self,
+        api_key: str,
+        environment: str
+    ):
+        # Pinecone can't accept quotes in either parameters, and these are sometimes
+        # unintentionally included in env variables
+        # We quit before initializing with a more informative error message
+        if "'" in api_key or '"' in api_key:
+            raise ValueError("Pinecone `api_key` contains single or double quotes, which isn't allowed.")
+        if "'" in environment or '"' in environment:
+            raise ValueError("Pinecone `environment` contains single or double quotes, which isn't allowed.")
+
+        pinecone.init(
+            api_key=api_key,
+            environment=environment,
+        )
+
+    def create_collection(self, schema: Type[VectorSchemaBase]):
+        collection_name = self.transform_collection_name(schema.collection_name())
+        current_indexes = pinecone.list_indexes()
+        if collection_name in current_indexes:
+            info("Collection already created...")
+            return
+
+        info("Creating collection, this could take 30s-5min...")
```

Comment:
I'd print timer seconds if you're going to warn user of time it takes to build so they can understand if they are seeing typical or atypical build times